src/clinical/api/clinical-api.ts

```diff
@@ -186,15 +183,56 @@ class ClinicalController {
 
 		const donorEntityData = await service.getDonorEntityData(donorIds);
 
+		const allExceptions: ExceptionManifestRecord[][] = [];
+
 		const date = currentDateFormatted();
 		const fileName = `filename=Donor_Clinical_Data_${date}.zip`;
+
+		if (donorEntityData.length) {
+			const entityRecords = donorEntityData[0].records;
+
+			// File Table can request multiple programs
+			const donorPrograms = entityRecords
+				.map((record) => record.program_id)
+				.filter(
+					// Remove duplicate program names
+					(programId, index, array): programId is string =>
+						typeof programId === 'string' && array.indexOf(programId) === index,
+				);
+
+			for (const programShortName of donorPrograms) {
+				// Get Donor Ids + Exceptions for each program
+				const programDonors = entityRecords
+					.filter(
+						(record) =>
+							typeof record.donor_id === 'number' && record.program_id === programShortName,
+					)
+					.map((record) => Number(record.donor_id))
```

Comment:
This `Number()` conversion is no longer necessary.
```suggestion
					.filter(
						(record): record is (ClinicalInfo & {donor_id: number, program_id: string}) =>
							typeof record.donor_id === 'number' && record.program_id === programShortName,
					)
					.map((record) => record.donor_id)
```